### ISSUE TITLE
2D HDR makes scenes significantly darker

### ISSUE THREAD

**Issue by ArseniyMirniy** [godotengine/godot#98708]
### Tested versions

Godot 4.3 Stable

### System information

Windows 10 x64, RTX 3060 Ti, Ryzen 5600x

### Issue description

Enabling “HDR” makes everything (and especially HDRi) significantly darker. I've tried several methods to locate any relevant side of the issue (physical light units, auto exposure, etc). Nothing changes that, it's just right away darker. No screen shaders are applied.

![image](https://github.com/user-attachments/assets/21994fcb-b192-4fdf-8ff8-ad12d4b0d23c)

### Steps to reproduce

Select 3D scene
Enable 2D HDR

### Minimal reproduction project (MRP)

N/A

**Comment by clayjohn** [comment]
Can you provide more information and an MRP?

For example, what is in your scene? How can someone else reproduce this issue? What renderer are you using? What environment settings do you have? What format is the texture you are using? are you drawing it with a shader? or a built in material?

Ideally you would provide an MRP that reproduces the issue so contributors can reproduce the issue and investigate a solution

**Comment by ArseniyMirniy** [comment]
> Can you provide more information and an MRP?
> 
> For example, what is in your scene? How can someone else reproduce this issue? What renderer are you using? What environment settings do you have? What format is the texture you are using? are you drawing it with a shader? or a built in material?
> 
> Ideally you would provide an MRP that reproduces the issue so contributors can reproduce the issue and investigate a solution

1. Render is Forward+
2. My scene consists of Environment + EXR HDR ([link](https://polyhaven.com/a/klippad_sunrise_2)) + Camera
3. No shaders, no canvas items, no meshes over the camera
4. The darkening affects the main viewport of Godot engine.
5. To reproduce, just use ANYTHING in 3D scene, meshes are also getting darker
6. I was using ACES, but it can be reproduced with any mode
7. I was using auto exposure, but on the shot below it's disabled

### Here's linear colors mode in the environment + sphere with StandardMaterial3D
![image](https://github.com/user-attachments/assets/d3b2bd75-4f7c-417b-b7f0-a1caca106f91)

**Comment by clayjohn** [comment]
Did you restart the editor after turning on HDR 2D?

**Comment by ArseniyMirniy** [comment]
> Thank you for your response! That really helps.
> 
> One more question:
> 
> Did you restart the editor after turning on HDR 2D?

Yes, since to apply 2D HDR you have to click on Save and Restart the editor

**Comment by allenwp** [comment]
I believe this issue is related to having the Environment set to use a Background of "Clear Color":

![Image](https://github.com/user-attachments/assets/2a08ec31-57ed-4591-bba0-5f606b9e1297)

If I were to take a guess at the root issue here, I would say the sRGB vs. linear encoding of `rendering/environment/defaults/default_clear_color` is being incorrectly interpreted when HDR 2D is enabled. Specifically, it seems that the sRGB-encoded `Color` of Default Clear Color is being treated as if it's linear-encoded and is being converted to sRGB, meaning that the `linear_to_sRGB` function is being applied when it shouldn't be. If I'm correct, the fix to this issue would be to make `rendering/environment/defaults/default_clear_color` always be correctly interpreted as an sRGB-encoded `Color`, regardless of whether HDR 2D is enabled or not.

As stated in the comments, this is not specific to an HDR EXR image; any mesh will do. Here's a MRP for testing:

[2D-HDR-ClearColor-bug.zip](https://github.com/user-attachments/files/20798650/2D-HDR-ClearColor-bug.zip)

| Environment Background Mode | HDR 2D Disabled | HDR 2D Enabled
| --- | --- | ---
Clear Color | ![Image](https://github.com/user-attachments/assets/ee756940-8938-436e-b8ba-bc5ac8cd3178) | ![Image](https://github.com/user-attachments/assets/1435fccd-79fc-4330-95ee-c7c17b7d208e)
Custom Color | ![Image](https://github.com/user-attachments/assets/b784628a-3e1b-491e-8836-daa2636b904b) | ![Image](https://github.com/user-attachments/assets/9ff7d431-7aae-43ef-b365-46f3e99eb24d)

**Comment by allenwp** [comment]
To those subscribed to this thread, I’ve put together a fix that should fully resolve this issue: #108682 

Feel free to try out the fix using the build artifacts on the Checks page of the PR and let me know if it resolves the issues for you.
